packages/page-accounts/src/util.tsx

```diff
@@ -27,23 +28,91 @@ function expandList (mapped: SortedAccount[], entry: SortedAccount): SortedAccou
   return mapped;
 }
 
-export function sortAccounts (addresses: string[], favorites: string[]): SortedAccount[] {
+export type AccountIdIsh = AccountId | AccountIndex | Address | string | Uint8Array | null;
+
+export function GetAccountCryptoType (accountId: AccountIdIsh): string {
+  try {
+    const current = accountId
+      ? keyring.getPair(accountId.toString())
+      : null;
+
+    if (current) {
+      return current.meta.isInjected
+        ? 'injected'
+        : current.meta.isHardware
+          ? current.meta.hardwareType as string || 'hardware'
+          : current.meta.isExternal
+            ? current.meta.isMultisig
+              ? 'multisig'
+              : current.meta.isProxied
+                ? 'proxied'
+                : 'external'
+            : current.type;
+    }
+  } catch {
+    // cannot determine, keep unknown
+  }
+
+  return 'unknown';
+}
+
+// Javascript `Array.prototype.sort` is not stable,
+// but stable sort is preffered for UI.
+// This function does not sort inplace.
+function stableSort<T> (elements: T[], cmp: (a: T, b: T) => number): T[] {
+  const indexedElements: [T, number][] = elements.map((x, index) => [x, index]);
+  const stableCmp = (a: [T, number], b: [T, number]) =>
+    cmp(a[0], b[0]) || a[1] - b[1];
+
+  indexedElements.sort(stableCmp);
+
+  return indexedElements.map(([x]) => x);
+}
```

Comment:
this is Accounts-free/agnostic. What about extracting to a sep util file?
benefits: + sharing in other areas, + more explicit -> easier to replace with some other algorithm in the future if needed.

Reply:
what about testing this little functionality? just a bunch of basic ones?

Reply:
Sure. Moving this away into a separate file and adding some tests.